### PR TITLE
Image component

### PR DESCRIPTION
**Pull request by Sundiata2**
Simple image component we'll use to build avatar then user-pills

screen shot 2018-05-08 at 3 25 25 pm

### CODE REVIEW

File: src/scss/views/widgets/_image.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.image-component {
```
Comment (alexander-daniel) [inline src/scss/views/widgets/_image.scss]: `component` is sorta implicit. let's call this something specific like
`circle-image` or something (i think that's what it is doing?)

Comment (Sundiata2) [reply src/scss/views/widgets/_image.scss]: We could just call it image-wrapper, it's literally just a wrapper for images

File: src/tests/views/widgets/image.test.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import test from 'ava';
+import { shallow } from 'enzyme';
+import Image from '../../../ts/views/widgets/image';
+
+test('Image gets classnames properly and img recieves ', (t) => {
+  const wrapper = shallow(<Image url={`http://via.placeholder.com/350x150`} className="test-image"/>);
```
Comment (alexander-daniel) [inline src/tests/views/widgets/image.test.tsx]: pass in some custom style too

File: src/tests/views/widgets/image.test.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  t.is(wrapper.hasClass('image-component test-image'), true);
+});
```
Comment (alexander-daniel) [inline src/tests/views/widgets/image.test.tsx]: write another test where you don't pass in a `url` and assert that resulting `style` is as expected (aka no `backgroundImage` style is set)

File: src/tests/views/widgets/image.test.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import test from 'ava';
+import { shallow } from 'enzyme';
+import Image from '../../../ts/views/widgets/image';
+
+test('Image gets classnames properly and img recieves ', (t) => {
+  const wrapper = shallow(<Image
+    url={`http://via.placeholder.com/350x150`}
```
Comment (alexander-daniel) [inline src/tests/views/widgets/image.test.tsx]: can be regular string

File: src/tests/views/widgets/image.test.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    className="test-image"
+    height={150}
+    width={350}/>);
```
Comment (alexander-daniel) [inline src/tests/views/widgets/image.test.tsx]: 🍗  put the `)` on next line

File: src/tests/views/widgets/image.test.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import test from 'ava';
+import { shallow } from 'enzyme';
+import Image from '../../../ts/views/widgets/image';
+
+test('Image gets classnames properly and img recieves ', (t) => {
+  const wrapper = shallow(<Image
```
Comment (alexander-daniel) [inline src/tests/views/widgets/image.test.tsx]: and put `<image` on the next line, as you would in a `return (` jsx thing

File: src/ts/views/widgets/image.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    className,
+    style,
+    url,
+    height,
+    width
+  } = props;
+
+  const imageClass = classnames('image-wrapper', className);
+  const customStyle = style || {
+    height: height ? `${height}px` : '150px',
```
Comment (alexander-daniel) [inline src/ts/views/widgets/image.tsx]: generic defaults? why not ensure you have to pass in height and width

File: src/ts/views/widgets/image.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    width: width ? `${width}px` : '200px'
+  };
+  const backgroundStyle = { backgroundImage: `url(${url})` };
+
+  return (
+    <div
+      className={imageClass}
+      style={Object.assign(customStyle, backgroundStyle)}
+      >
```
Comment (alexander-daniel) [inline src/ts/views/widgets/image.tsx]: de-dent one

**Comment by alexander-daniel** [comment]
few 🍗 then :truck:!!

**Pull request merged**